homeassistant/components/zwave_me/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""The Z-Wave-Me WS integration."""
+import logging
+
+from zwave_me_ws import ZWaveMe, ZWaveMeData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect, dispatcher_send
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, PLATFORMS, ZWAVEPLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Z-Wave-Me from a config entry."""
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Comment:
You should verify the connection can be established before setting up the platforms. That way the platforms don't need to do this check.

Reply:
@MartinHjelmare suggested to always set up platforms and not from the connection:

> Since we set up all platforms regardless of the device added, I’d recommend setting up the platforms from async_setup_entry from the beginning. Then we can call the async API hass.config_entries.async_setup_platforms directly since we’re in async context inside the event loop. There’s no point in waiting for the first device if we don’t differentiate set up platforms depending on the device added.

Reply:
There is a difference between first device arriving and verifying that you are able to establish a connection. If the server is down now, you will never notify the user.